jams/core.py

```diff
@@ -309,7 +309,83 @@ def __len__(self):
 
     def __repr__(self):
         """Render the object alongside its attributes."""
-        return '<{}: {:s}>'.format(self.type, ', '.join(self.keys()))
+        indent = len(self.type) + 2
+        jstr = ',\n' + ' ' * indent
+
+        props = self._display_properties()
+
+        params = jstr.join('{:}={:}'.format(p, summary(self[p],
+                                                       indent=indent))
+                           for (p, dp) in props)
+        return '<{}({:})>'.format(self.type, params)
+
+    def _display_properties(self):
+        '''Returns a list of tuples (key, display_name)
+        for properties of this object'''
+
+        return sorted([(k, k) for k in self.__dict__])
+
+    def _repr_html_(self):
+
+        props = self._display_properties()
+
+        if not props:
+            return ''
+
+        out = '<div class="panel-group">'
+        for (prop, dprop) in props:
+            content = summary_html(self[prop])
+
+            prop_class = 'default'
+            if not content:
+                prop_class = 'danger'
+
+            out += '<div class="panel panel-{}">'.format(prop_class)
+
+            if (isinstance(self[prop], (JObject, AnnotationArray, dict))
+               and content):
```

Comment:
and yes, it passes pycodestyle and pylint only complains about redefining eval and some import shenanigans.